packages/block-editor/src/components/block-list/block.js

```diff
@@ -728,6 +730,7 @@ function BlockListBlockProvider( props ) {
 		themeSupportsLayout,
 		isTemporarilyEditingAsBlocks,
 		blockEditingMode,
+		editorMode,
```

Comment:
As we're touching the deps, we should monitor the editor performance after merging this PR.

Reply:
I was remembering https://github.com/WordPress/gutenberg/pull/56994 about combining block store subscriptions, and assumed it would be better to add it here rather than check on each block. Maybe the performance potential isn't worth it though. I do like the implementation though, as it makes the final result quite clean.

Reply:
That's a good point about combining store subscriptions. Let's try it "as is" and watch the performance charts post-merge.

Reply:
I did some smoke test with running performance in the inspector and entering into/out of zoom out mode and this method of using the combined subscription and not adding listeners was always faster in both total scripting and rendering, so I say we go with it as well.

Reply:
Great. I say we merge it and then watch the performance dashboard in case of any changes post-merge.